### ISSUE TITLE
Let modules manage their own symbol in their parent scope

### ISSUE THREAD

**Issue by sergisiso** [stfc/PSyclone#2729]
With #2596 Routines manage their symbol in the parent scope (attaching, detaching, renaming the routines affects their scope), modules should have the same behaviour. @LonelyCat124 I guess we can use the same Node functionality now?

**Comment by LonelyCat124** [comment]
Yeah - I assume we can use an `update_parent_symbol_table` implementation to do something with this.
